xmtp_mls/migrations/2023-10-29-205333_state_machine_init/up.sql

```diff
@@ -0,0 +1,72 @@
+CREATE TABLE groups (
+    -- Random ID generated by group creator
+    "id" BLOB PRIMARY KEY NOT NULL,
+    -- Based on the timestamp of the welcome message
+    "created_at_ns" BIGINT NOT NULL,
+    -- Enum of GROUP_MEMBERSHIP_STATE
+    "membership_state" INT NOT NULL
+);
+
+-- Allow for efficient sorting of groups
+CREATE INDEX groups_created_at_idx ON groups(created_at_ns);
+
+CREATE INDEX groups_membership_state ON groups(membership_state);
+
+-- Successfully processed messages meant to be returned to the user
+CREATE TABLE group_messages (
+    "id" BLOB PRIMARY KEY NOT NULL,
+    -- Derived via SHA256(CONCAT(decrypted_message_bytes, conversation_id, timestamp))
+    "group_id" BLOB NOT NULL,
+    -- Message contents after decryption
+    "decrypted_message_bytes" BLOB NOT NULL,
+    -- Based on the timestamp of the message
+    "sent_at_ns" BIGINT NOT NULL,
+    -- Enum GROUP_MESSAGE_KIND
+    "kind" INT NOT NULL,
+    -- Could remove this if we added a table mapping installation_ids to wallet addresses
```

Comment:
I would be in favor of this - we probably will need something eventually to track validity/revocation/expiries

Reply:
I'm still on the fence about this. It's a lot simpler to just rely on the MLS group state and the network to tell you when something has changed or is expired. I think a lot of the data we'd be tracking there would be duplicative. But I haven't dug in too deeply. I'd suggest we leave it in the "maybe" column until we're closer to implementation or actually have a concrete need for it.